src/main/java/br/edu/ufcg/computacao/alumni/core/MatchesFinder.java

```diff
@@ -82,12 +82,18 @@ public static synchronized MatchesFinder getInstance() {
 				count++;
 			}
 		}
+		
 		if (count == splitedFilteredName.length) return splitedFilteredName;
+		
 		String[] resizedSplitedFilteredName = new String[count];
-		for (int i = 0; i < count; i++) {
-			resizedSplitedFilteredName[i] = splitedFilteredName[i];
+		int index = 0;
+		for (String filteredName : splitedFilteredName) {
+			if (filteredName != null) {
```

Comment:
Esse teste deveria ser feito na hora que está construindo splitedFilteredName (linha 81). Caso contrário, se tiver algum campo null, o length de resizedSplitedFilteredName vai ficar errado. Isso impacta em outros lugares da lógica.

Reply:
Acho que não, pois antes de fazer a operação que gera o splitedFilteredName, nós não sabemos o tamanho do vetor final (sem os conectivos), por isso que, por exemplo: se eu colocar "jonatas ferreira de lima", o splitedFilteredName é ["JONATAS", "FERREIRA", null, "LIMA"], e o segundo cria um novo vetor do tamanho do contador do primeiro for (que é o tamanho real do nome) e retira o null do vetor

Reply:
Na realidade, esse teste nem deveria existir. Basta que a linha 81 seja:
splitedFilteredName[count++] = namePart;

E remove a linha 82. Além disso, o loop das linhas 90-94 deveria ser:
for (int i = 0; i < count; i++) {
       resizedSplitedFilteredName[i] = splitedFilteredName[i];
}

Reply:
Entendi entendi. Testei aqui manualmente e tá saindo certo, vou implementar.